packages/tpetra/core/test/MultiVector/MultiVector_UnitTests.cpp

```diff
@@ -5366,6 +5362,103 @@ namespace {
 #endif
 
 
+
+  TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL( MultiVector, syncTest_replaceGlobalValues, LO , GO , Scalar , Node ) {
+    using namespace Tpetra::Details;
+    typedef Tpetra::Map<LO, GO, Node> map_type;
+    typedef Tpetra::MultiVector<Scalar,LO, GO, Node> MV;
+    typedef typename MV::impl_scalar_type IST;
+    IST IST_one = Teuchos::ScalarTraits<IST>::one();
+    Scalar S_one = Teuchos::ScalarTraits<Scalar>::one();
+
+    // Make vector
+    RCP<const Comm<int> > comm = Tpetra::getDefaultComm ();
+    RCP<const map_type> map = rcp (new map_type (100, 0, comm));
+    MV x(map, 1);
+
+    // Names and such
+    auto exec_space = typename Node::execution_space();
+    const std::string space = exec_space.name();
+    const bool is_gpu = Spaces::is_gpu_exec_space<typename Node::execution_space>();
+    size_t global_count, instance_count;
+
+    // NOTE: Global fences (aka Kokkos::fence()) fence both the host and device execution space,
+    // so our counts there get doubled.
+
+    /***********************************************************************/
+    // Device-then-replace (impl_scalar_type)
+    out<<"*** Device-then-replace impl_scalar_type ***"<<std::endl;
+    {
+      {
+        auto x_d = x.getLocalViewDevice(Tpetra::Access::ReadWrite);
+      }
+      Tpetra::Details::FenceCounter::reset();
+      Tpetra::Details::FenceCounter::start();
+      x.replaceGlobalValue(map->getGlobalElement(0),0,IST_one);
+      Tpetra::Details::FenceCounter::stop();
+      global_count = Tpetra::Details::FenceCounter::get_count_global(space);
+      instance_count = Tpetra::Details::FenceCounter::get_count_instance(space);
+      TEST_EQUALITY(global_count,2*is_gpu);
+      TEST_EQUALITY(instance_count,0);
```

Comment:
Is the way to read this that if on a GPU, there are 2 fences somewhere, but not in `replaceGlobalValue`?